watertap/examples/flowsheets/crystallization/multi_effect_crystallizer.py

```diff
@@ -0,0 +1,306 @@
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+)
+from idaes.core.util.testing import initialization_tester
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+    badly_scaled_var_generator,
+)
+from idaes.core import UnitModelCostingBlock
+
+from watertap.costing import WaterTAPCosting, CrystallizerCostType
+
+solver = get_solver()
+
+
+def build_multi_effect_crystallizer_fs(
+    m=None,
+    operating_pressure_eff1=0.35,  # bar
+    operating_pressure_eff2=0.208,  # bar
+    operating_pressure_eff3=0.095,  # bar
+    feed_mass_frac_NaCl=0.2126,
+    feed_pressure=101325,  # Pa
+    feed_temperature=273.15 + 20,  # K
+    crystallizer_yield=0.5,
+):
+    """
+    This flowsheet depicts a 3-effect crystallizer, with brine fed in parallel
+    to each effect, and the operating pressure is specfied individually.
+    """
+
+    if m is None:
+        m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.fs.props = props.NaClParameterBlock()
+
+    eff_1 = m.fs.effect_1 = Crystallization(property_package=m.fs.props)
+    eff_2 = m.fs.effect_2 = Crystallization(property_package=m.fs.props)
+    eff_3 = m.fs.effect_3 = Crystallization(property_package=m.fs.props)
+
+    # Specify the feed properties
+    feed_mass_frac_NaCl = 0.2126
+    feed_mass_frac_H2O = 1 - feed_mass_frac_NaCl
+    eps = 1e-6
+
+    eff_1.inlet.pressure[0].fix(feed_pressure)
+    eff_1.inlet.temperature[0].fix(feed_temperature)
+    eff_1.inlet.flow_mass_phase_comp[0, "Sol", "NaCl"].fix(eps)
+    eff_1.inlet.flow_mass_phase_comp[0, "Vap", "H2O"].fix(eps)
+
+    eff_2.inlet.pressure[0].fix(feed_pressure)
+    eff_2.inlet.temperature[0].fix(feed_temperature)
+    eff_2.inlet.flow_mass_phase_comp[0, "Sol", "NaCl"].fix(eps)
+    eff_2.inlet.flow_mass_phase_comp[0, "Vap", "H2O"].fix(eps)
+
+    eff_3.inlet.pressure[0].fix(feed_pressure)
+    eff_3.inlet.temperature[0].fix(feed_temperature)
+    eff_3.inlet.flow_mass_phase_comp[0, "Sol", "NaCl"].fix(eps)
+    eff_3.inlet.flow_mass_phase_comp[0, "Vap", "H2O"].fix(eps)
+
+    # Set up for effect 1
+    # Operating pressure: 0.35 bar
+    c1_pressure = 101325 * operating_pressure_eff1
+    feed_flow_mass = 1
+
+    eff_1.inlet.flow_mass_phase_comp[0, "Liq", "NaCl"].fix(
+        feed_flow_mass * feed_mass_frac_NaCl
+    )
+    eff_1.inlet.flow_mass_phase_comp[0, "Liq", "H2O"].fix(
+        feed_flow_mass * feed_mass_frac_H2O
+    )
+
+    eff_1.pressure_operating.fix(c1_pressure)
+    eff_1.crystallization_yield["NaCl"].fix(crystallizer_yield)
+
+    eff_1.crystal_growth_rate.fix()
+    eff_1.souders_brown_constant.fix()
+    eff_1.crystal_median_length.fix()
+
+    # Set up for effect 2
+    # Same crystallizer_yield
+    # Operating pressure: 0.208 bar
+    c2_pressure = 101325 * operating_pressure_eff2
+
+    eff_2.pressure_operating.fix(c2_pressure)
```

Comment:
you could just do 
`eff_2.pressure_operating.fix(0.28 * pyunits.bar)`

(assuming you imported `from pyomo.environ import units as pyunits`)

Also, it looks like you multiply each operating pressure by 101325 Pa, but the conversion is technically 1e5 Pa/ bar so wouldn't your effective operating pressures be slightly higher than intended?

Reply:
Yes you're right. I'll correct it, thanks!